sdk/spring/scripts/ci_update_versions.py

```diff
@@ -0,0 +1,190 @@
+                for line in newlines:
+                    f.write(line)
+
+    except Exception as e:
+        print("Unexpected exception: " + str(e))
+        traceback.print_exc(file=sys.stderr)
+
+def load_version_map_from_file(the_file, version_map):
+    with open(the_file) as f:
+        for raw_line in f:
+            stripped_line = raw_line.strip()
+            if not stripped_line or stripped_line.startswith('#'):
+                continue
+            module = CodeModule(stripped_line)
+            # verify no duplicate entries
+            if (module.name in version_map):
+                raise ValueError('Version file: {0} contains a duplicate entry: {1}'.format(the_file, module.name))
+            # verify that if the module is beta_ or unreleased_ that there's a matching non-beta_ or non-unreleased_ entry
+            if (module.name.startswith('beta_') or module.name.startswith('unreleased_')):
+                tempName = module.name
+                if tempName.startswith('beta_'):
+                    tempName = module.name[len('beta_'):]
+                else:
+                    tempName = module.name[len('unreleased_'):]
+                # if there isn't a non beta or unreleased entry then raise an issue
+                if tempName not in version_map:
+                    raise ValueError('Version file: {0} does not contain a non-beta or non-unreleased entry for beta_/unreleased_ library: {1}'.format(the_file, module.name))
+
+            version_map[module.name] = module
+
+def display_version_info(version_map):
+    for value in version_map.values():
+        print(value)
+
+def update_versions_all(update_type, target_file):
+    # dictionary
+    version_map = {}
+    ext_dep_map = {}
+    # Load the version of external dependency file for the given UpdateType into the version_map.
+    if update_type == UpdateType.external_dependency:
+        dependency_file = SPRING_BOOT_MANAGED_EXTERNAL_DEPENDENCIES_FILE_NAME
+        print('external_dependency_file=' + dependency_file)
+        load_version_map_from_file(dependency_file, ext_dep_map)
+
+    display_version_info(version_map)
+    display_version_info(ext_dep_map)
+
+    if target_file:
+        update_versions(update_type, version_map, ext_dep_map, target_file)
+    else:
+        for root, _, files in os.walk("./sdk/spring"):
+            for file_name in files:
+                file_path = root + os.sep + file_name
+                if  file_name.startswith('pom') and file_name.endswith('.xml'):
+                    update_versions(update_type, version_map, ext_dep_map, file_path)
+
+def main():
+    parser = argparse.ArgumentParser(description='Replace version numbers in poms.')
+    parser.add_argument('--update-type', '--ut', type=UpdateType, choices=list(UpdateType))
+    parser.add_argument('--target-file', '--tf', nargs='?', help='File to update (optional) - all files in the current directory and subdirectories are scanned if omitted')
```

Comment:
> All files in the current directory

Not `current directory.`


We do not need to update specific file. How about do like this:

```
    parser.add_argument(
        '--targer-folder',
        type=str,
        required=False,
        default='./sdk/string',
        help='Set target folder.'
    )
```